lib/Reference/TableReferenceProvider.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace OCA\Tables\Reference;
+
+use Exception;
+use OC\Collaboration\Reference\LinkReferenceProvider;
+use OCA\Tables\Service\TableService;
+use OCP\Collaboration\Reference\IReferenceProvider;
+use OCP\Collaboration\Reference\Reference;
+use OC\Collaboration\Reference\ReferenceManager;
+use OCA\Tables\AppInfo\Application;
+use OCP\Collaboration\Reference\IReference;
+use OCP\IURLGenerator;
+use Throwable;
+
+class TableReferenceProvider implements IReferenceProvider {
+	private const RICH_OBJECT_TYPE = Application::APP_ID . '_table';
+
+	private ?string $userId;
+	private ReferenceManager $referenceManager;
+	private IURLGenerator $urlGenerator;
+	private LinkReferenceProvider $linkReferenceProvider;
+	private TableService $tableService;
+
+	public function __construct(IURLGenerator $urlGenerator,
+								TableService $tableService,
+								ReferenceManager $referenceManager,
+								LinkReferenceProvider $linkReferenceProvider,
+								?string $userId) {
+		$this->userId = $userId;
+		$this->referenceManager = $referenceManager;
+		$this->urlGenerator = $urlGenerator;
+		$this->linkReferenceProvider = $linkReferenceProvider;
+		$this->tableService = $tableService;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function matchReference(string $referenceText): bool {
+		if ($this->userId === null) {
+			return false;
+		}
+		$start = $this->urlGenerator->getAbsoluteURL('/apps/' . Application::APP_ID);
```

Comment:
Might be nice to just reuse the getTableIdFromLink method here since the matching logic should be similar.

Reply:
:+1: